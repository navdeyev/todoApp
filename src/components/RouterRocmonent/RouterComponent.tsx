import {History} from 'history';
import * as React from 'react';
import {Route, Router} from 'react-router';

import PageLayout from 'components/PageLayout/PageLayout';
import TodosContainer from 'components/TodosContainer/TodosContainer';

export interface IProps {
    history: History
}

const RouterComponent: React.SFC<IProps> = (props) => {
    const {history} = props;
    return (
        <Router history={history}>
            <Route path="/" component={PageLayout}>
                <Route path="todos" component={TodosContainer}/>
            </Route>
        </Router>
    );
};

export default RouterComponent;