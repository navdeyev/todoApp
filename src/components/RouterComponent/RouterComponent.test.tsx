import {shallow} from 'enzyme';
import {History} from 'history';
import * as React from 'react';

import RouterComponent, {IProps} from './RouterComponent';

describe('RouterComponent', () => {

    it('renders a component', () => {
        const defaultProps: IProps = {
            history: {} as History
        };

        const router = shallow(<RouterComponent {...defaultProps} />);
        expect(router.getElement()).toMatchSnapshot();
    });

});
