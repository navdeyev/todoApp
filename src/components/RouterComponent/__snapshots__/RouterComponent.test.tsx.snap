// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterComponent renders a component 1`] = `
<Router
  history={Object {}}
  render={[Function]}
>
  <Redirect
    from="/"
    to="todos"
  />
  <Route
    component={[Function]}
    path="/"
  >
    <Route
      component={[Function]}
      path="todos"
    />
  </Route>
</Router>
`;
